### DISCUSSION TITLE
APIs to remove in v7

### DISCUSSION THREAD

**Discussion by RDIL** [thepeacockproject/Peacock#346] [Ideas]
Hey all,

Opening this to suggest some redundant/problematic APIs we might want to get rid of in v7.

## `MasteryService#getMasteryDataForDestination`
<img width="341" alt="Screenshot 2023-11-25 at 13 17 38" src="https://github.com/thepeacockproject/Peacock/assets/34555510/1a5a9ecb-46e3-46f3-8ca6-796b06e3fddc">

This API is literally identical to `MasteryService#getMasteryData` - it just forwards the call there. This seems pointless to have.

## `LogCategory#CALLER`

<img width="461" alt="Screenshot 2023-11-25 at 13 19 48" src="https://github.com/thepeacockproject/Peacock/assets/34555510/1a503ed4-5a99-44cb-83f1-13bdc3b02c62">

Using `arguments.caller` is highly problematic, and doesn't even work in strict mode. It's also deprecated, see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/caller).

Let me know what everyone thinks.

**Comment by AnthonyFuller** [comment]
The destination function was there for a reason as at one point it was different. I think maybe MVM removed the need for it though.
